### DISCUSSION TITLE
about beginFrame and endFrame

### DISCUSSION THREAD

**Discussion by JackKaiXing** [google/filament#4223] [Q&A] [answered]
Hello, 

I am wondering  does the problem, described below, is possible to occasionally occur in filament?

___________

1. we are loading GLTF asset with `ResourceLoader::AsyncBeginLoad`.

2. after we get `!(asyncGetLoadProgress < 1.0)`, we add the GLTF drawable to filament scene.

3. we do, only once, `filament::Renderer::beginFrame` and `filament::Renderer::endFrame`.

Does it possible, the GLTF model could not been rendered because of just one frame?

Thanks. And we are using filament-1.9.18, we have not updated it for our project.

**Comment by prideout** [comment]
If you use `asyncBeginLoad` rather than `loadResources`, then you also need to call `asyncUpdateLoad` before calling beginFrame.

**Comment by JackKaiXing** [comment]
I do call `asyncUpdateLoad` until` !(asyncGetLoadProgress < 1.0)  == true`.

**Reply by JackKaiXing** [reply]
@prideout any ideas？

**Comment by JackKaiXing** [comment, marked as answer]
beginframe return false, it is skipped.

**Reply by prideout** [reply]
If you want to avoid using our frame skipper, you are allowed to ignore the `beginFrame` return code.  You can simply unconditionally render the view if you'd like.
